fdbserver/DataDistribution.actor.cpp

```diff
@@ -5252,12 +5260,19 @@ ACTOR Future<Void> dataDistribution(Reference<DataDistributorData> self,
 			primaryTeamCollection = Reference<DDTeamCollection>();
 			remoteTeamCollection = Reference<DDTeamCollection>();
 			wait(shards.clearAsync());
-			if (err.code() != error_code_movekeys_conflict)
-				throw err;
-			bool ddEnabled = wait(isDataDistributionEnabled(cx, ddEnabledState));
-			TraceEvent("DataDistributionMoveKeysConflict").detail("DataDistributionEnabled", ddEnabled).error(err);
-			if (ddEnabled)
-				throw err;
+			TraceEvent("DataDistributorTeamCollectionsDestroyed").error(err);
+			if (removeFailedServer.getFuture().isReady() && !removeFailedServer.getFuture().isError()) {
+				TraceEvent("RemoveFailedServer", removeFailedServer.getFuture().get()).error(err);
+				wait(removeKeysFromFailedServer(cx, removeFailedServer.getFuture().get(), lock, ddEnabledState));
+				wait(removeStorageServer(cx, removeFailedServer.getFuture().get(), lock, ddEnabledState));
+			} else {
+				if (err.code() != error_code_movekeys_conflict)
+					throw err;
```

Comment:
Style nitpick: I think we should use `if (err.code() != error_code_movekeys_conflict) throw err;` (single line) or 
`if (err.code() != error_code_movekeys_conflict) {`
                                         `throw err;`
`}` braces for multi-line